src/main/java/com/bhoruka/bloodbank/web/CampController.java

```diff
@@ -0,0 +1,57 @@
+package com.bhoruka.bloodbank.web;
+
+import com.bhoruka.bloodbank.model.request.CreateCampRequest;
+import com.bhoruka.bloodbank.model.response.CreateCampResponse;
+import com.bhoruka.bloodbank.service.CampService;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/camp")
+@Slf4j
+public class CampController {
+
+    private static final String CAMP_CREATE_SUCCESS_MESSAGE = "Camp created successfully.";
+
+    @NonNull
+    private CampService campService;
+
+    /**
+     * Constructor for the camp controller.
+     * @param campService mandatory instance of CampService
+     */
+    @Autowired
+    public CampController(@NonNull final CampService campService) {
+        this.campService = campService;
+    }
+
+    /**
+     * Controller endpoint for creating a new camp.
+     * @param createCampRequest the incoming create request
+     * @return the response with the newly created camp id
+     */
+    @PostMapping("/create")
+    public CreateCampResponse createCamp(@RequestBody final CreateCampRequest createCampRequest) {
+        log.info("Received create request : {}", createCampRequest);
+
+        CreateCampResponse response = null;
+        String campId = campService.createCamp(createCampRequest);
```

Comment:
Would Optional help with the null check here? (not sure)

Reply:
Yes, we could use optional instead of a null check and then throw a custom exception in case the optional is empty.